gapic/src/main/com/google/gapid/util/Logging.java

```diff
@@ -123,6 +135,7 @@ public static void setListener(Runnable listener) {
    */
   public static void logMessage(Log.Message message) {
```

Comment:
I've separated `Buffer` and `Handler`. The inner class was doing too much - hence the extraction of the event logic. Hopefully this is now cleaner and would be nice and simple to make the classes public if needed in the future.